### PR TITLE
[doc] fix outdated java api examples

### LINKED ISSUE #9976: Docs: Are there outdated APIs  (e.g., DateHistogram) in Java API docs?
I am checking the APIs in Java API docs, and it seems that I found outdated APIs. I hope some contributors may confirm whether the followings are outdated or not: 

1. QueryBuilders.queryString
https://github.com/elasticsearch/elasticsearch/blob/master/docs/java-api/search.asciidoc. The page includes the following code snippet:
MultiSearch API …

```java
SearchRequestBuilder srb1 = node.client()
     .prepareSearch().setQuery(QueryBuilders.queryString("elasticsearch")).setSize(1);
```
==> I found that the QueryBuilders class does not have the queryString method.

2. DateHistogram
https://github.com/elasticsearch/elasticsearch/blob/master/docs/java-api/aggs.asciidoc. The page includes the following code snippet:
Structuring aggregations …

```java
SearchResponse sr = node.client().prepareSearch()     
   .addAggregation(               
      AggregationBuilders.terms("by_country").field("country")         
      .subAggregation(AggregationBuilders.dateHistogram("by_year")             
      .field("dateOfBirth")             
      .interval((DateHistogram.Interval.YEAR)             
 .subAggregation(AggregationBuilders.avg("avg_children").field("children"))         )     
)     
.execute().actionGet();
```

==> I found that the DateHistogram class was deleted in Jan. 27th, this year. For the class, the following pages also should be checked: https://github.com/elasticsearch/elasticsearch/blob/master/docs/java-api/aggregations/bucket/datehistogram-aggregation.asciidoc

3. DateRange
https://github.com/elasticsearch/elasticsearch/blob/master/docs/java-api/aggregations/bucket/daterange-aggregation.asciidoc.
Use aggregation response
Import Aggregation definition classes:

```java
import org.elasticsearch.search.aggregations.bucket.range.date.DateRange;
// sr is here your SearchResponse object 
DateRange agg = sr.getAggregations().get("agg");  
// For each entry 
for (DateRange.Bucket entry : agg.getBuckets()) {     
   String key = entry.getKey();                    
   // Date range as key     
   DateTime fromAsDate = entry.getFromAsDate();    
   // Date bucket from as a Date     
   DateTime toAsDate = entry.getToAsDate();        
   // Date bucket to as a Date     
   long docCount = entry.getDocCount();            
   // Doc count      
   logger.info("key [{}], from [{}], to [{}], doc_count [{}]", key, fromAsDate, toAsDate, docCount); }
```

==> I found that the DateRange class was deleted in Jan. 27th, this year, and does not exist anymore. 

4. IPv4Range
https://github.com/elasticsearch/elasticsearch/blob/master/docs/java-api/aggregations/bucket/iprange-aggregation.asciidoc
Use aggregation response
Import Aggregation definition classes:

```java
import org.elasticsearch.search.aggregations.bucket.range.ipv4.IPv4Range;
// sr is here your SearchResponse object 
IPv4Range agg = sr.getAggregations().get("agg");  
// For each entry 
for (IPv4Range.Bucket entry : agg.getBuckets()) {
     String key = entry.getKey();                    
     // Ip range as key     
     String fromAsString = entry.getFromAsString();  
     // Ip bucket from as a String     
     String toAsString = entry.getToAsString();      
     // Ip bucket to as a String     
     long docCount = entry.getDocCount();            
     // Doc count      
     logger.info("key [{}], from [{}], to [{}], doc_count [{}]", key, fromAsString, toAsString, docCount); }
```

==> I found that the IPv4Range class was also deleted in Jan. 27th, this year. 

Thanks in advance,
Seonah

### PR DESCRIPTION
**Pull request by dadoonet**
Some API have been changed in master. Master documentation should reflect be updated:
- QueryBuilders.queryString is now QueryBuilders.queryStringQuery
- DateHistogram.Interval is now DateHistogramInterval
- Refactoring of buckets in aggs
- FilterBuilders has been replaced by QueryBuilders

Closes #9976.

### CODE REVIEW

File: docs/java-api/aggregations/bucket/geodistance-aggregation.asciidoc [hunk 1]
```diff
@@ -40,10 +40,10 @@ Range agg = sr.getAggregations().get("agg");
 
 // For each entry
 for (Range.Bucket entry : agg.getBuckets()) {
-    String key = entry.getKey();                     // key as String
+    String key = (String) entry.getKey();            // key as String
```
Comment (rjernst) [inline docs/java-api/aggregations/bucket/geodistance-aggregation.asciidoc]: getKeyAsString()?

**Comment by rjernst** [comment]
LGTM

**Comment by dpursehouse** [comment]
Note: There are still several references to `FilterBuilders`, which has been removed, in docs/java-api/query-dsl-filters.asciidoc

**Comment by dadoonet** [comment]
Thank you @dpursehouse for raising it. I started revisiting the JAVA API documentation about this. Stay tuned :)

**Comment by dadoonet** [comment]
@dpursehouse I pushed some changes in master branch for the documentation. Don't hesitate to raise other issues or sending PR if there are other things that should be fixed. Thanks!

**Comment by dpursehouse** [comment]
Thanks @dadoonet.  If I find anything else I'll try to fix it myself.

BTW do you have any idea when there will be a release candidate for 2.0.0?

**Pull request merged**